src/libraries/Common/src/System/Buffers/Text/Base64Encoder.cs

```diff
@@ -15,8 +16,29 @@ namespace System.Buffers.Text
     /// <summary>
     /// Convert between binary data and UTF-8 encoded text that is represented in base 64.
     /// </summary>
-    public static partial class Base64
+#if SYSTEM_PRIVATE_CORELIB
+    internal
+#else
+    public
+#endif
+        static partial class Base64
     {
+        // This can be replaced once https://github.com/dotnet/runtime/issues/63331 is implemented.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<byte> SimdShuffle(Vector128<byte> left, Vector128<byte> right, Vector128<byte> mask8F)
+        {
+            Debug.Assert((Ssse3.IsSupported || AdvSimd.Arm64.IsSupported) && BitConverter.IsLittleEndian);
+
+            if (Ssse3.IsSupported)
+            {
+                return Ssse3.Shuffle(left, right);
+            }
+            else
+            {
+                return AdvSimd.Arm64.VectorTableLookup(left, Vector128.BitwiseAnd(right, mask8F));
+            }
+        }
```

Comment:
Can/should this be the implementation of the new API(s) rather than having a one-off here?

Reply:
Changed code that I don't touch this function any more.
There is one more place where a similar "temp" helper is used - I'll check and clean up in a separate PR, ok?